azure-nextgen-cs-appservice/AppServiceStack.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.
+
+using Pulumi;
+using Pulumi.AzureNextGen.Resources.Latest;
+using Pulumi.AzureNextGen.Web.Latest;
+using Pulumi.AzureNextGen.Web.Latest.Inputs;
+using Pulumi.AzureNextGen.Insights.Latest;
+using Pulumi.AzureNextGen.Sql.Latest;
+using Pulumi.AzureNextGen.Storage.Latest;
+using Pulumi.AzureNextGen.Storage.Latest.Inputs;
+
+class AppServiceStack : Stack
+{
+    public AppServiceStack()
+    {
+        var resourceGroup = new ResourceGroup("appservice-rg", new ResourceGroupArgs
+        {
+            ResourceGroupName = "appservice-rg",
+            Location = "westus2",
+        });
+
+        var storageAccount = new StorageAccount("sa", new StorageAccountArgs
+        {
+            AccountName = "appservicesa",
+            ResourceGroupName = resourceGroup.Name,
+            Kind = "StorageV2",
+            Sku = new SkuArgs
+            {
+                Name = SkuName.Standard_LRS,
+            },
+        });
+
+        var appServicePlan = new AppServicePlan("asp", new AppServicePlanArgs
+        {
+            Name = "asp",
+            Location = resourceGroup.Location,
+            ResourceGroupName = resourceGroup.Name,
+            Kind = "App",
+            Sku = new SkuDescriptionArgs
+            {
+                Tier = "Basic",
+                Name = "B1",
+            },
+        });
+
+        var container = new BlobContainer("zips", new BlobContainerArgs
+        {
+            AccountName = storageAccount.Name,
+            ContainerName = "zips",
+            PublicAccess = PublicAccess.None,
+            ResourceGroupName = resourceGroup.Name,
+        });
+
+        var blob = new Blob("zip", new BlobArgs
+        {
+            BlobName = "appservice-blob",
+            ResourceGroupName = resourceGroup.Name,
+            AccountName = storageAccount.Name,
+            ContainerName = container.Name,
+            Type = BlobType.Block,
+            Source = new FileArchive("wwwroot"),
+        });
+
+        var codeBlobUrl = signedBlobReadUrl(blob, container, storageAccount, resourceGroup);
+        
+        this.SignedURL = codeBlobUrl;
+
+        var appInsights = new Component("appInsights", new ComponentArgs
+        {
+            Location = resourceGroup.Location,
+            ApplicationType = "web",
+            Kind = "web",
+            ResourceGroupName = resourceGroup.Name,
+            ResourceName = "appInsights",
+        });
+
+        var config = new Config();
+        var username = config.Get("sqlAdmin") ?? "pulumi";
+        var password = config.RequireSecret("sqlPassword");
```

Comment:
We could also generate a random password (not critical, I guess you copied this from another example)